internal/pkg/template/input.go

```diff
@@ -0,0 +1,77 @@
+package template
+
+import (
+	"reflect"
+
+	goValidator "github.com/go-playground/validator/v10"
+	"github.com/umisama/go-regexpcache"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/utils/orderedmap"
+	"github.com/keboola/keboola-as-code/internal/pkg/validator"
+)
+
+type Inputs []*Input
+
+type Input struct {
+	Id          string                 `json:"id" validate:"required,template-input-id"`
+	Name        string                 `json:"name" validate:"required"`
+	Description string                 `json:"description" validate:"required"`
+	Type        string                 `json:"type" validate:"required"`
+	Default     interface{}            `json:"default,omitempty" validate:"omitempty,template-input-default"`
+	Kind        string                 `json:"kind" validate:"required,oneof=input password textarea confirm select multiselect"`
+	Options     []Option               `json:"options,omitempty" validate:"required_if=Type select Type multiselect,template-input-options,dive,template-input-option"`
+	Rules       *orderedmap.OrderedMap `json:"rules,omitempty"`
```

Comment:
Moze byt `string` ako navrhujes v Jire, dobry postreh.

Reply:
No já si nebyl jistej jak to má vlastně vypadat. Ale imho pole stringů který si potom pro validaci spojíme do jednoho se bude zapisovat přehledněji.

Reply:
No ... ale este je tam ten `OR` co mi doslo az potom, znak `|` ... ako som pisal v Jire.

Reply:
No jo, to je pravda, tím se to zkomplikuje. 🤔 Ok, tak zatím string. 🙂